From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Emilia Schotte <pancake@mgnet.work>
Date: Sat, 29 Nov 2025 08:52:56 +0100
Subject: [PATCH 3/3] net: ethernet: mtk_eth_soc: Fix immutable string IRQ
 setup

The PDMA IRQ setup code uses a string literal pointer for the ring name,
which is mutated shortly after.

Change rxring to a character array as to not write to immutable-memory.

Signed-off-by: Emilia Schotte <pancake@mgnet.work>
---
 drivers/net/ethernet/mediatek/mtk_eth_soc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mediatek/mtk_eth_soc.c b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
index 0d76af26445a..2481aa2a8485 100644
--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
@@ -3773,7 +3773,7 @@ static int mtk_get_irqs_fe(struct platform_device *pdev, struct mtk_eth *eth)
 
 static int mtk_get_irqs_pdma(struct platform_device *pdev, struct mtk_eth *eth)
 {
-	char *rxring = "pdma0";
+	char rxring[] = "pdma0";
 	int i;
 
 	for (i = 0; i < MTK_PDMA_IRQ_NUM; i++) {
